hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/utils/ContainerCache.java

```diff
@@ -227,9 +246,68 @@ private boolean cleanupDb(ReferenceCountedDB db) {
   public void addDB(String containerDBPath, ReferenceCountedDB db) {
     lock.lock();
     try {
-      this.putIfAbsent(containerDBPath, db);
+      openContainerCache.putIfAbsent(containerDBPath, db);
+      metrics.incNumOpenContainerInserts();
+    } finally {
+      lock.unlock();
+    }
+  }
+
+  public void markContainerClosed(String containerPath) {
+    lock.lock();
+    try {
+      ReferenceCountedDB db = openContainerCache.remove(containerPath);
+      Preconditions.checkNotNull(db, "Container not found in open cache:" +
+          containerPath);
+      closeContainerCache.putIfAbsent(containerPath, db);
+      metrics.incNumOpenContainerRemoves();
     } finally {
       lock.unlock();
     }
   }
+
+  public boolean isContainerOpen(State state) {
```

Comment:
Would a better separation between the container states be `isContainerWritable`? Open is becoming an over loaded term.

Reply:
ok will do that,

Reply:
done in the followup patch